MapboxNavigation/FeedbackViewController.swift

```diff
@@ -70,6 +108,69 @@ class FeedbackViewController: UIViewController {
         dismissFeedback()
     }
     
+    func didTapCell(_ sender: UIGestureRecognizer) {
+        guard sender.state == .began || sender.state == .ended else { return }
+        
+        abortAutodismiss()
+        
+        let touchLocation = sender.location(in: self.collectionView)
+        
+        if let indexPath = self.collectionView.indexPathForItem(at: touchLocation) {
+            // get the cell at indexPath (the one you long pressed)
+            let cell = self.collectionView.cellForItem(at: indexPath) as! FeedbackCollectionViewCell
+            cell.layer.add(pulsingAnimation, forKey: "animateScale")
+            
+            activeFeedbackItem = sections[indexPath.section][indexPath.row]
+            
+            
+            if audioRecorder == nil {
+                startRecording()
+            }
+            
+            if sender.state == .ended {
+                finishRecording()
+            }
+        }
+    }
+    
+    func audioRecorderDidFinishRecording(_ recorder: AVAudioRecorder, successfully flag: Bool) {
+        if let fileData = NSData(contentsOfFile: recorder.url.path) as Data? {
+            activeFeedbackItem!.audio = fileData
+            sendFeedbackHandler?(activeFeedbackItem!)
+        }
+    }
+    
+    
+    func startRecording() {
+        let audioFile = getDocumentsDirectory().appendingPathComponent("recording-\(Date().timeIntervalSince1970).m4a")
+        
+        let settings = [
+            AVFormatIDKey: Int(kAudioFormatMPEG4AAC),
+            AVSampleRateKey: 12000,
+            AVNumberOfChannelsKey: 1,
+            AVEncoderAudioQualityKey: AVAudioQuality.low.rawValue
+        ]
+        
+        do {
+            audioRecorder = try AVAudioRecorder(url: audioFile, settings: settings)
+            audioRecorder?.delegate = self
+            audioRecorder?.record()
+        } catch {
+            print(error.localizedDescription)
+        }
+    }
+    
+    func finishRecording() {
+        audioRecorder?.stop()
+        audioRecorder = nil
+    }
+    
+    func getDocumentsDirectory() -> URL {
+        let paths = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
```

Comment:
We shouldn't write to the app’s document directory. How about `cachesDirectory`?